src/main/java/frc/robot/subsystems/NoteVision.java

```diff
@@ -111,6 +125,24 @@ public List<Pose2d> getNotes(Pose2d pose) {
         return poses;
     }
 
+    public boolean checkForNote(DriveTrain driveTrain, int wantedNote, NoteVision noteVision) {
```

Comment:
Kind of confused. No need to pass in NoteVision; you are in it.
Also, you should just pass in the Pose2d (or Translate2d) of the thing you are looking for, and the use that.